lib/ddtrace/profiling/recorder.rb

```diff
@@ -19,11 +19,16 @@ def push(event)
         @buffers[event.class].push(event)
       end
 
-      def pop(event_class)
-        raise UnknownEventError, event_class unless @buffers.key?(event_class)
-        @buffers[event_class].pop
+      def pop
+        @buffers.collect do |event_class, buffer|
+          events = buffer.pop
+          next if events.empty?
+          Flush.new(event_class, events)
+        end.compact
       end
 
+      Flush = Struct.new(:event_class, :events).freeze
```

Comment:
Found using a `Hash` was much less space efficient, so opted for a struct to pair type with events instead.